src/test/java/io/strimzi/kafka/metrics/KafkaMetricsCollectorTest.java

```diff
@@ -51,10 +50,15 @@ public void testCollect() {
         metrics = collector.collect();
         assertTrue(metrics.isEmpty());
 
-        // Adding a non-numeric metric does nothing
-        collector.addMetric(buildNonNumericMetric("name2", "group"));
+        // Adding a non-numeric metric converted
+        KafkaMetric nonNumericMetric = buildNonNumericMetric("name", "group");
+        collector.addMetric(nonNumericMetric);
         metrics = collector.collect();
-        assertTrue(metrics.isEmpty());
+        assertEquals(1, metrics.size());
+        assertEquals("kafka_server_group_name", metrics.get(0).name);
+        assertEquals(1, metrics.get(0).samples.size());
```

Comment:
extract variable to avoid the repetition.

Reply:
Or do you mean something like:
```
List<Collector.MetricFamilySamples.Sample> samples = metrics.get(0).samples;
        assertEquals(1, metrics.size());
        assertEquals("kafka_server_group_name", metrics.get(0).name);
        assertEquals(1, samples.size());
        assertEquals(1.0, samples.get(0).value, 0);
        assertTrue(samples.get(0).labelNames.contains("value"));
        }

Reply:
I mean extract `metrics.get(0)` as a variable.

Reply:
Thanks

Reply:
Addressed